<a href="https://colab.research.google.com/github/Ishikaaa/ML-Data-Scientist-job-market-in-the-US/blob/master/Data_Scientist_job_market_in_the_US.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***1. Importing Libraries - ***

In [0]:
import pandas as pd 
import numpy as np 

In [0]:
dataset=pd.read_csv("alldata.csv")
dataset.head()

position  ...            location
0                               Development Director  ...  Atlanta, GA 30301 
1  An Ostentatiously-Excitable Principal Research...  ...         Atlanta, GA
2                                     Data Scientist  ...         Atlanta, GA
3                                       Data Analyst  ...  Atlanta, GA 30303 
4  Assistant Professor -TT - Signal Processing & ...  ...         Atlanta, GA

[5 rows x 5 columns]

In [0]:
dataset.shape 

(7183, 5)

In [0]:
dataset.dtypes

position        object
company         object
description     object
reviews        float64
location        object
dtype: object

# ***Handling Missing Data - ***

In [0]:
dataset.isnull().sum()

position         11
company          11
description      11
reviews        1660
location         11
dtype: int64

In [0]:
# removing that rows in which all colunms are null 
dataset = dataset.dropna(how='all',axis=0)
dataset.shape

(7172, 5)

In [0]:
dataset.isnull().sum()

position          0
company           0
description       0
reviews        1649
location          0
dtype: int64

In [0]:
# remove that rows which do not have any review 
dataset=dataset.fillna(0)
dataset.shape 

(7172, 5)

In [0]:
dataset.isnull().sum()

position       0
company        0
description    0
reviews        0
location       0
dtype: int64

In [0]:
dataset['position'].value_counts()

Data Scientist                                                     355
Senior Data Scientist                                               97
Research Analyst                                                    64
Data Engineer                                                       60
Machine Learning Engineer                                           56
Lead Data Scientist                                                 31
Research Scientist                                                  30
Sr. Data Scientist                                                  28
Principal Data Scientist                                            24
Software Engineer                                                   22
Data Analyst                                                        22
Quantitative Analyst                                                21
Scientist                                                           17
Product Manager                                                     17
Bioinf

In [0]:
dataset['company'].value_counts()

Amazon.com                                           358
Ball Aerospace                                       348
Google                                               138
Microsoft                                            137
NYU Langone Health                                    77
Fred Hutchinson Cancer Research Center                70
KPMG                                                  66
Lab126                                                50
Broad Institute                                       49
Facebook                                              49
Walmart eCommerce                                     45
McKinsey & Company                                    41
Harnham                                               39
Takeda Pharmaceutical                                 39
University of Washington                              39
Pharmacyclics, an Abbvie Company                      34
Oath Inc                                              34
Celgene Corporation            

In [0]:
dataset['reviews'].value_counts()

0.0        1649
108.0       354
25902.0     224
2.0         191
5.0         139
4.0         133
4618.0      118
2474.0      110
3.0         101
6.0          94
25895.0      89
7.0          79
705.0        76
91.0         71
8.0          68
15.0         66
838.0        62
32.0         55
11.0         54
366.0        49
14.0         48
141.0        46
12.0         46
70.0         42
26.0         42
10.0         41
385.0        41
530.0        39
52.0         39
17.0         36
           ... 
3578.0        1
242.0         1
222.0         1
2203.0        1
25909.0       1
2189.0        1
415.0         1
694.0         1
723.0         1
312.0         1
1516.0        1
55.0          1
49.0          1
3027.0        1
11607.0       1
1947.0        1
131.0         1
177.0         1
63.0          1
228.0         1
892.0         1
2431.0        1
2647.0        1
447.0         1
1959.0        1
6868.0        1
1366.0        1
21949.0       1
290.0         1
2409.0        1
Name: reviews, Length: 5

In [0]:
dataset['description'][0]

"Development Director\nALS Therapy Development Institute has an immediate opening for Development Directors. Reporting directly to the Senior Development Director, the Development Director at ALS TDI is a senior fundraising position working to identifying potential prospects and cultivating solicitation strategies and in closing asks with donors including individuals and corporations by building networks via events, generating awareness of ALS TDI; outreach including attending and speaking at events as well as personally cultivates relationships with patients, prospects and donors. This position will be responsible for generating and managing a portfolio of at least two million to five million dollars per year. This position will be located in Atlanta, GA.\n\nRequirements:\nBachelor's Degree requiredMinimum 6-8 years' experience in fundraising and business developmentSuccessful track recording in fundraising with major donors or scientific sales preferredDemonstrated ability to work in

In [0]:
dataset['location'].value_counts()

Seattle, WA                  563
New York, NY                 508
Cambridge, MA                487
Boston, MA                   454
Boulder, CO                  435
San Francisco, CA            425
San Diego, CA                294
Chicago, IL                  292
Washington, DC               230
Sunnyvale, CA                221
Mountain View, CA            199
Los Angeles, CA              187
Austin, TX                   156
Atlanta, GA                  147
Redmond, WA 98052            117
Seattle, WA 98109             95
Cambridge, MA 02139           95
New York, NY 10016            83
San Diego, CA 92121           71
Cambridge, MA 02142           64
South San Francisco, CA       53
Chicago, IL 60654             45
Chicago, IL 60601             40
San Francisco, CA 94103       40
Redmond, WA                   39
Mountain View, CA 94043       39
Seattle, WA 98101             34
Seattle, WA 98104             34
Cambridge, MA 02140           33
Boston, MA 02116              31
          

***Note - ***

1.   Most of the job offers are on the location 'Seattle, WA'.



In [0]:
dataset[dataset['reviews']==dataset['reviews'].max()]

position  ...             location
6947  Staff Software Engineer, Front End Lead  ...  Sunnyvale, CA 94086
7047         Information Security Risk Expert  ...  Sunnyvale, CA 94086
7177         Senior Data Scientist (GEC11902)  ...  Sunnyvale, CA 94086

[3 rows x 5 columns]

In [0]:
dataset[dataset['company']=='Walmart']

position  ...             location
449                        Data Scientist-ISD  ...     Austin, TX 78716
6887                     Staff Data Scientist  ...  San Bruno, CA 94066
6947  Staff Software Engineer, Front End Lead  ...  Sunnyvale, CA 94086
7047         Information Security Risk Expert  ...  Sunnyvale, CA 94086
7177         Senior Data Scientist (GEC11902)  ...  Sunnyvale, CA 94086

[5 rows x 5 columns]

***Note - ***

1.   Company "Walmart" has maximum reviews. And Company "Walmart" offers maximum jobs on location "Sunnyvale, CA 94086".

In [0]:
dataset[dataset['reviews']==dataset['reviews'].min()]

position  ...             location
0                                  Development Director  ...   Atlanta, GA 30301 
1     An Ostentatiously-Excitable Principal Research...  ...          Atlanta, GA
2                                        Data Scientist  ...          Atlanta, GA
18                                       Data Scientist  ...          Atlanta, GA
19                                       Data Scientist  ...          Atlanta, GA
25                                       Data Scientist  ...          Atlanta, GA
26                 Data Scientist 1 – AP Forensics Team  ...          Atlanta, GA
38                                       Data Scientist  ...          Atlanta, GA
48                        Chief Deep Learning Scientist  ...          Atlanta, GA
53                                  Statistical Analyst  ...          Atlanta, GA
54                                   VP, Data Analytics  ...          Atlanta, GA
55                              Lead Decision Scientist  ...   Atlanta, GA 30313 
65                                   Research Scientist  ...          Atlanta, GA
84                                   Research Scientist  ...          Atlanta, GA
106   Staff Research Scientist VII: Informational In...  ...          Atlanta, GA
114                             Jr Microbiologist - CDC  ...          Atlanta, GA
122               Mid Data Scientist - Machine Learning  ...   Atlanta, GA 30342 
139   Principal Research Scientist I: Informational ...  ...          Atlanta, GA
140                             Market Research Analyst  ...          Atlanta, GA
151                        Senior Scientist, Immunology  ...          Atlanta, GA
153          A Vigorously-Charged Scientific Programmer  ...          Atlanta, GA
154                                   Software Engineer  ...          Atlanta, GA
156                                Director of Software  ...   Atlanta, GA 30332 
157       Certifying Scientist / Quantitation Scientist  ...          Atlanta, GA
161                                      Data Scientist  ...   Atlanta, GA 30326 
164   Windows Programmer / Software Engineer - Inter...  ...          Atlanta, GA
170   Windows Programmer / Software Engineer - Entry...  ...          Atlanta, GA
192   Various Professional, Technical, Administrativ...  ...          Atlanta, GA
196     Business Development Manager (Southeast Region)  ...          Atlanta, GA
199   Implementation Engineer (Machine Learning/Data...  ...   Atlanta, GA 30326 
...                                                 ...  ...                  ...
7008                                     Head of Design  ...  Sunnyvale, CA 94085
7010                 Forward Deployed Solution Engineer  ...        Sunnyvale, CA
7014        Deep Learning Software Development Engineer  ...        Sunnyvale, CA
7019                                 Knowledge Engineer  ...        Sunnyvale, CA
7032  AI Research/Software Development Engineer, Ale...  ...        Sunnyvale, CA
7034                         3D Tools Software Engineer  ...  Sunnyvale, CA 94089
7035       Software Engineer - Tools and Infrastructure  ...        Sunnyvale, CA
7036  23103 Principal Data Scientist - HealthTech / ...  ...        Sunnyvale, CA
7041                                      Manager, TPMs  ...        Sunnyvale, CA
7043                               QA Software Engineer  ...        Sunnyvale, CA
7044                                Sr Speech Scientist  ...        Sunnyvale, CA
7046                      Software Development Engineer  ...        Sunnyvale, CA
7048                                  Software Engineer  ...        Sunnyvale, CA
7060                      Sr. Machine Learning Engineer  ...        Sunnyvale, CA
7065                        Sr. Hardware Audio Engineer  ...        Sunnyvale, CA
7067                       Materials (Quality) Engineer  ...  Sunnyvale, CA 94086
7075  Senior Data Scientist, Algorithms and Machine ...  ...        Sunnyvale, CA
7087                    Senio